app/layout_elements.py

```diff
@@ -82,6 +83,9 @@ def display_costs(
     with st.expander("**Data**"):
         column_config = config_number_columns(df_res, format=f"%.1f {output_unit}")
         st.dataframe(df_res, use_container_width=True, column_config=column_config)
+        fn = f"costs_per_{key}_{key_suffix}".strip("_")

```

Comment:
For costs, I suggest to add the content of ``select_data`` to the filename to indicate whether it contains results based on default data, modified data, or the difference between them. 

```suggestion
        fn = f"costs_per_{key}_{key_suffix}".strip("_")
        if st.session_state["user_changes_df"] is not None:
            fn = f"{fn}_{select_data}".replace(" ", "_")
```